libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -1238,47 +1383,46 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
     /**
      * @private
      */
-    private removeOverlappingPerProperty(candidates: Partial<EntityAssignment>[]): Partial<EntityAssignment>[] {
+    private removeOverlappingPerProperty(candidates: EntityAssignment[]): EntityAssignment[] {
         // Group mappings by property
-        const perProperty = candidates.reduce<{ [path: string]: Partial<EntityAssignment>[] }>(
-            (accumulator, assignment): {} => {
-                if (accumulator.hasOwnProperty(assignment.property)) {
-                    accumulator[assignment.property].push(assignment);
-                } else {
-                    accumulator[assignment.property] = [assignment];
-                }
-                return accumulator;
-            },
-            {}
-        );
-
-        const output: Partial<EntityAssignment>[] = [];
-        for (const path in perProperty) {
-            const entityPreferences = this.entityPreferences(path);
-            let choices = perProperty[path];
-
-            // Assume preference by order listed in mappings
-            // Alternatives would be to look at coverage or other metrics
-            for (const entity of entityPreferences) {
-                let candidate: Partial<EntityAssignment>;
-                do {
-                    candidate = undefined;
-                    for (let i = 0; i < choices.length; i++) {
-                        const mapping = choices[i];
-                        if (mapping.entity.name == entity) {
-                            candidate = mapping;
-                            break;
+        const perProperty = candidates.reduce<{ [path: string]: EntityAssignment[] }>((accumulator, assignment): Record<
```

Comment:
`reduce` is generally good about inferring types, are these explicit type params necessary? If anything, you should only have to provide `reduce<T>((acc, curr) => ..., defaultValue);` and the rest is inferred.

Reply:
Not necessary. Removed.